### PR TITLE
Align KTO with DPO: Add tests

### PR DESCRIPTION
**Pull request by albertvillanova**
Align KTO with DPO: Add tests.

This PR aligns the experimental KTO trainer tests with the DPO trainer tests by expanding coverage and standardizing test names.

Part of:
- #4786

### Changes
- Rename tests to follow the DPO naming convention (test_kto_trainer → test_train, test_kto_trainer_with_ref_model_is_model → test_train_with_ref_model_is_model_raises, etc.)
- Reorder tests to match the DPO test file order
- Add new tests mirroring DPO test coverage: test_train_gpt_oss, test_train_with_explicit_ref_model, test_train_model_dtype, test_train_peft_model, test_train_with_peft_config_and_gradient_checkpointing, test_init_fails_with_peft_and_liger, test_train_with_iterable_dataset, test_train_with_chat_template_kwargs, test_train_toolcall_data, test_train_with_eval, test_train_with_multiple_eval_dataset, test_train_with_gradient_checkpointing, test_tag_added, test_tag_added_peft, test_peft_with_quantization
- Improve existing test assertions using descriptive failure messages and torch.testing.assert_close where appropriate

---

> [!NOTE]
> **Low Risk**
> Test-only changes to experimental KTO coverage; no production trainer logic is modified.
> 
> **Overview**
> Expands and reorganizes **experimental KTO trainer** tests in `test_kto_trainer.py` so they mirror the DPO trainer test suite—same naming, order, and breadth of scenarios—without changing `KTOTrainer` implementation.
> 
> Renames the main parametrized smoke test and related cases from `test_kto_trainer_*` to the DPO-style `test_train*` / `test_init_fails_*` pattern, and reorders methods to match the DPO file layout. Adds many new integration tests: Harmony/GPT-OSS training, explicit vs synced reference models, `model_init_kwargs` dtype, PEFT (including gradient checkpointing and 4-bit quantization via `BitsAndBytesConfig`), Liger kernel and the PEFT+Liger init error, streaming datasets with `apo_zero_unpaired`, per-example `chat_template_kwargs`, tool-call data, single and multi eval datasets, gradient-checkpointing regression guards, and `kto`/`trl` model tags.
> 
> Tightens assertions with parameter-name failure messages and `torch.testing.assert_close` where the reference model must stay frozen. Imports are updated (`require_bitsandbytes`, conditional `peft` helpers) to support the new cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 0692b7fe443b9489f21f95513ea71585844b1b1e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: tests/experimental/test_kto_trainer.py [hunk 1]
```diff
@@ -587,47 +622,422 @@ def test_kto_trainer_with_liger(self):
...
+
+        # `chat_template_kwargs` is applied per example: the prepared prompts must contain both the lowercase and the
+        # uppercase system header, proving the argument was threaded through tokenization (the dataset alternates
+        # `role_capital`). The exact row order is not asserted because preprocessing unpairs the dataset.
+        decoded_prompts = [
+            trainer.processing_class.decode(trainer.train_dataset[i]["prompt_ids"])
+            for i in range(len(trainer.train_dataset))
+        ]
+        assert any("<|im_start|>SYSTEM" in prompt for prompt in decoded_prompts)
+        assert any("<|im_start|>system" in prompt for prompt in decoded_prompts)
```
[the remark this answers is not in the document]
Comment (albertvillanova) [reply tests/experimental/test_kto_trainer.py:872]: I think this needs to be addressed on a separate PR:
- https://github.com/huggingface/trl/pull/6161

**Review by qgallouedec — APPROVED** [review]
